### DISCUSSION TITLE
Queston: Is it possible to unit test named job proccess handler calls with jest?

### DISCUSSION THREAD

**Discussion by ealcantara22** [OptimalBits/bull#2010] [Q&A] [answered]
## Description
Hello, I would like to know if its possible to test named job processor calls after call the queue `add` method using jest. Also, an example of how to accomplish this would be great. 
## Minimal, Working Test code to reproduce the issue.
Looking to accomplish something like:

```javascript
// worker.js

const redisHost = require('my-redis-host'),
  Queue = require('bull'),
  fruitQueue = new Queue('fruitQueue', redisHost);

// consumer handler
const fruitHandler = (job) => {
   const { fruitId } = job.data;
   // my logic
}

// consumer
fruitQueue.process('updateFruit', 1, fruitHandler);

module.exports = { fruitQueue, fruitHandler } 

```

```javascript
// worker.test.js

test(`Should call the fruitHandler`, () => {
  const worker = require('./worker'),
    job = {
      fruitId: 1
    };

   worker.fruitHandler = jest.fn(() => Promise.resolve({}));

    // calling the producer 
    fruitQueue.add('updateFruit', job);

   worker.fruitHandler).toHaveBeenCalledTimes(1)

   // also tried waiting 
   // setTimeout(expect(worker.updateInElasticById).toHaveBeenCalledTimes(1), 1000)
);

```

The response always is:
```
Error: expect(jest.fn()).toHaveBeenCalledTimes(expected)

Expected number of calls: 1
Received number of calls: 0
```

## Bull version
 3.20.1
## Additional information
I've already tried mocking the bull lib. Also, ioredis-mock following this [thread](https://github.com/OptimalBits/bull/issues/962) and of course using fully working redis instance up and running.

**Comment by eric573** [comment]
I am having the same issue as well. Any thoughts on this?

**Comment by younes-io** [comment]
I also have the same issue. Has anyone managed to find a solution to this ?

**Comment by manast** [comment, marked as answer]
As I see it you could just unit test the handler isolated from the process function, just make sure that you properly mock the "job" object that you pass to the handler.

**Comment by xtrinch** [comment]
Same issue here
